pkg/types/validation/installconfig_test.go

```diff
@@ -1208,6 +1219,58 @@ func TestValidateInstallConfig(t *testing.T) {
 			}(),
 			expectedError: `\Q[networking.machineNewtork[0]: Invalid value: "172.17.64.0/18": overlaps with default Docker Bridge subnet, platform: Invalid value: "libvirt": must specify one of the platforms (\E.*\Q)]\E`,
 		},
+		{
+			name: "with valid workloadSettings",
+			installConfig: func() *types.InstallConfig {
+				return validWorkloadSettings()
+			}(),
+			expectedError: ``,
+		},
+		{
+			name: "with invalid workloadSettings.name",
+			installConfig: func() *types.InstallConfig {
+				c := validWorkloadSettings()
+				c.Workload[0].Name = "badvalue"
+				return c
+			}(),
+			expectedError: `^workloadSettings\[0\].name: Unsupported value: "badvalue": supported values: .*`,
+		},
+		{
+			name: "with empty workloadSettings.name",
+			installConfig: func() *types.InstallConfig {
+				c := validWorkloadSettings()
+				c.Workload[0].Name = ""
+				return c
+			}(),
+			expectedError: `^workloadSettings\[0\].name: Unsupported value: "": supported values: .*`,
+		},
+		{
+			name: "with duplicate workloadSettings.name",
+			installConfig: func() *types.InstallConfig {
+				c := validWorkloadSettings()
+				c.Workload = append(c.Workload, types.WorkloadPartition{
+					Name:   types.ManagementWorkloadPartition,
+					CPUIds: "2-3",
+				})
+				return c
+			}(),
+			expectedError: `^workloadSettings\[1\].name: Duplicate value: "management"$`,
+		},
+		{
+			name: "with empty workloadSettings.cpuIDs",
+			installConfig: func() *types.InstallConfig {
+				c := validWorkloadSettings()
+				c.Workload[0].CPUIds = ""
+				return c
+			}(),
+			expectedError: `^workloadSettings\[0\].cpuIDs: Invalid value: "": cannot be empty`,
+		},
+	}
+	for _, id := range []string{"0", "0-1", "3,5,7", "3,5-8,11", "2-3,5-6,8-9"} {
+		cases = append(cases, validCpuidTestcase(id))
+	}
+	for _, id := range []string{"nodigits", "-7", "10-", ",2", "4,", "5,,6", "8--10", "2,4,bad", "5,6-,8", "2-3,-6,8-9", "10-6", "0,5-2,7"} {
+		cases = append(cases, invalidCpuidTestcase(id))
 	}
```

Comment:
These cases are testing the `cpuset.Parse` function, which is not the unit under test.

Reply:
This was a hold-over from an earlier invocation that used a regex for validation instead of the cpuset module like the current implementation does.  I left it in as proof that the other library does what we want and external change there doesn't break is in the future.  But I suppose the unit test is there in vendors/ if we need it.

Gone.